pkg/monitor/monitor/monitor_manager.go

```diff
@@ -427,3 +422,253 @@ func (m *MonitorManager) patchTidbClusterStatus(tcRef *v1alpha1.TidbClusterRef,
+		return false, err
+	}
+
+	if deploymentPv.Spec.ClaimRef != nil && deploymentPv.Spec.ClaimRef.Name == firstStsPvcName {
+		// smooth migration successfully and clean status
+		monitor.Status.DeploymentStorageStatus = nil
+		return true, nil
+	}
+
+	// firstly patch status
+	if monitor.Status.DeploymentStorageStatus == nil {
+		err = m.patchTidbMonitorStatus(monitor, deploymentPvc.Spec.VolumeName)
+		if err != nil {
+			return false, err
+		}
+		//monitor patch status successfully
+	}
+
+	err = m.deps.PVCControl.DeletePVC(monitor, &corev1.PersistentVolumeClaim{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      deploymentPvcName,
+			Namespace: monitor.Namespace,
+		},
+	})
+	if err != nil {
+		klog.Errorf("Fail to delete the PVC %s for tm [%s/%s], err: %v", deploymentPvcName, monitor.Namespace, monitor.Name, err)
+		return false, err
+	}
+	err = m.patchPVClaimRef(deploymentPv, firstStsPvcName, monitor)
+	if err != nil {
+		klog.Errorf("Smooth migration for tm[%s/%s], fail to patch PV %s, err: %v", monitor.Namespace, monitor.Name, deploymentPvc.Spec.VolumeName, err)
+		return false, err
+	}
+	// smooth migration successfully and clean status
+	monitor.Status.DeploymentStorageStatus = nil
+	return true, nil
+}
+
+func (c *MonitorManager) validate(tidbmonitor *v1alpha1.TidbMonitor) bool {
+	errs := v1alpha1validation.ValidateTidbMonitor(tidbmonitor)
+	if len(errs) > 0 {
+		aggregatedErr := errs.ToAggregate()
+		klog.Errorf("tidbmonitor %s/%s is not valid and must be fixed first, aggregated error: %v", tidbmonitor.GetNamespace(), tidbmonitor.GetName(), aggregatedErr)
+		c.deps.Recorder.Event(tidbmonitor, corev1.EventTypeWarning, "FailedValidation", aggregatedErr.Error())
+		return false
+	}
+	return true
+}
+
+func (m *MonitorManager) syncTidbMonitorPV(tm *v1alpha1.TidbMonitor) error {
+	ns := tm.GetNamespace()
+	instanceName := tm.Name
+
+	l, err := label.New().Instance(instanceName).TiDBMonitor().Selector()
+	if err != nil {
+		return err
+	}
+	pods, err := m.deps.PodLister.Pods(ns).List(l)
+	if err != nil {
+		return fmt.Errorf("fail to list pods for tidbmonitor %s/%s, selector: %s, error: %v", ns, instanceName, l, err)
+	}
+
+	for _, pod := range pods {
+		if component := pod.Labels[label.ComponentLabelKey]; component != label.TiDBMonitorVal {
+			// only update tidbmonitor pod
+			continue
+		}
+		// update meta info for pvc
+		pvcs, err := util.ResolvePVCFromPod(pod, m.deps.PVCLister)
+		if err != nil {
+			return err
+		}
+		for _, pvc := range pvcs {
+			if pvc.Spec.VolumeName == "" {
+				continue
+			}
+			// update meta info for pv
+			pv, err := m.deps.PVLister.Get(pvc.Spec.VolumeName)
+			if err != nil {
+				klog.Errorf("Get PV %s error: %v", pvc.Spec.VolumeName, err)
+				return err
+			}
+			_, err = m.deps.PVControl.UpdateMetaInfo(tm, pv)
+			if err != nil {
+				return err
+			}
+		}
+	}
+
+	return nil
+}
+
+func (m *MonitorManager) createOrUpdateService(newSvc *corev1.Service, monitor *v1alpha1.TidbMonitor) error {
```

Comment:
We can define this as a common function in util.go to share with other modules, right?

Reply:
No,i try to move this method to pkg/util package,it will cause package cycle call.

Reply:
What about make it a function in `pkg/member/manager` package just as you have done for `UpdateStatefulSet`.